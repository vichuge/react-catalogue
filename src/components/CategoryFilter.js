import React from 'react';
import PropTypes from 'prop-types';
import categories from '../containers/Categories';

const CategoryFilter = ({ categoryFilterChange }) => {
  const handleChange = (e) => {
    categoryFilterChange(e.target.value);
  };
  return (
    <select className="form-control mt-2" onChange={handleChange}>
      <option key="default" value="All">All books</option>
      {
        categories.map((cat) => <option key={cat} value={cat}>{cat}</option>)
      }
    </select>
  );
};

CategoryFilter.propTypes = {
  categoryFilterChange: PropTypes.func.isRequired,
};

export default CategoryFilter;
