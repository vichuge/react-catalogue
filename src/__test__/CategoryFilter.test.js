import React from 'react';
import renderer from 'react-test-renderer';
import CategoryFilter from '../components/CategoryFilter';

it('return CategoryFilter component', () => {
  const clickHandle = () => { };
  const tree = renderer
    .create(<CategoryFilter categoryFilterChange={clickHandle} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
