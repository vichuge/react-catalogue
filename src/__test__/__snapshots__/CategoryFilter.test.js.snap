// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return CategoryFilter component 1`] = `
<select
  className="form-control mt-2"
  onChange={[Function]}
>
  <option
    value="All"
  >
    All gen
  </option>
  <option
    value="gen-1"
  >
    gen-1
  </option>
  <option
    value="gen-2"
  >
    gen-2
  </option>
  <option
    value="gen-3"
  >
    gen-3
  </option>
</select>
`;
